source/cpp/source/ComponentSearch.cpp

```diff
@@ -86,11 +108,9 @@ juce::TopLevelWindow * ComponentSearch::findWindowWithId (const juce::String & i
     if (id.isEmpty ())
         return topWindows.front ();
 
-    auto it = std::find_if (
-        topWindows.begin (),
-        topWindows.end (),
-        [&] (auto && window)
-        { return window->getProperties ().getWithDefault (windowId, {}).toString () == id; });
+    auto it = std::find_if (topWindows.begin (),
+                            topWindows.end (),
+                            [&] (auto && window) { return windowHasId (*window, id); });
```

Comment:
Windows are components, any reason it needs a separate find?

Reply:
It uses a different property `window-id` instead of `component-id`. Arguably they could be the same, though 🤔